content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst

```diff
@@ -1,87 +1,106 @@
+
+- :guilabel:`Delivery Product` (*Required field*): the product listed on the :ref:`sales order line
+  <inventory/shipping/sales-order>` as the delivery charge.
+- :guilabel:`Free if order amount is above`: checking this box enables free shipping if the customer
+  spends above the specified amount.
+
+.. _inventory/shipping/invoice-so:
 
-The price is computed when you **save** the sale order or when you click on
-**Set price**.
+Invoice cost on sales order
+===========================
 
-To invoice the price of the delivery charge on the sale order, click on
-**Set price**, it will add a line with the name of the delivery method as
-a product. It may vary from the real price.
+To invoice shipping costs on the sales order, before the item is delivered, go to the
+:menuselection:`Sales app` and select the desired sales order.
 
-When you create the invoice, it will take the price set on the sale
-order.
+On the sales order, click the :guilabel:`Add Shipping` button at the bottom-right corner.
 
-.. image:: invoicing/invoicing01.png
+.. image:: invoicing/add-shipping.png
    :align: center
+   :alt: Click "Add Shipping" button at the bottom right, near the total.
 
-Invoice the real shipping costs
--------------------------------
+In the :guilabel:`Add a shipping method` pop-up window, choose the intended carrier in the
+:guilabel:`Shipping Method` field.
 
-On your sale order, choose the carrier that will be used. Click on
-**Delivery Method** to choose the right one.
+Then, click the :guilabel:`Get Rate` button to the calculate shipping price based on real-time
+shipping data Odoo's shipping carrier integration.
 
-.. image:: invoicing/invoicing05.png
+The :guilabel:`Cost` is automatically calculated using the weight of the items in the order.
+Finally, click the :guilabel:`Add` button to close the window.
+
+.. image:: invoicing/add-a-shipping-method.png
    :align: center
+   :alt: Calculate shipping by selecting a shipping method.
 
-The price is computed when you **save** the sale order. Confirm the sale
-order and proceed to deliver the product.
+.. _inventory/shipping/sales-order:
 
-The real shipping cost is computed when the delivery order is
-validated, you can see the real cost in the chatter of the delivery order.
+On the sales order, the delivery product appears in the :guilabel:`Order Lines` tab, with the
+:guilabel:`Unit Price` set as the shipping cost calculated in the :guilabel:`Add a shipping method`
+pop-up window.
 
-.. image:: invoicing/invoicing02.png
+.. image:: invoicing/delivery-product.png
    :align: center
+   :alt: Show delivery product on the sales order line.
 
-Go back to the sale order, the real cost is now added to the sale
-order.
+Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
+that includes the shipping cost that was added earlier.
 
-.. image:: invoicing/invoicing03.png
-   :align: center
+Invoice real shipping costs
+===========================
```

Comment:
Overall suggestion for this section is to add more UI walk-through. We *tell* more than *show* here and I'm confident this feedback would come up in KC's review anw 😉